custom_components/ble_monitor/__init__.py

```diff
@@ -513,9 +522,9 @@ def __init__(self, config, dataqueue):
         if self.config[CONF_DEVICES]:
             for device in self.config[CONF_DEVICES]:
                 if CONF_DEVICE_ENCRYPTION_KEY in device and device[CONF_DEVICE_ENCRYPTION_KEY]:
-                    p_mac = bytes.fromhex(device["mac"].replace(":", "").lower())
+                    p_id = bytes.fromhex(dict_get_or_clean(device).lower())
```

Comment:
Just that I understand it, can `p_id` either be a MAC address or an UUID?

Reply:
everything is right. there are types of beacon with encryption. I want to add more of them, but for now I added ibeacon as an example